File: Solutions/795. Number of Subarrays with Bounded Maximum/795. Number of Subarrays with Bounded Maximum.java
class Solution {
    public int numSubarrayBoundedMax(int[] nums, int L, int R) {
        int res=0,prev=-1,dp=0;
        for(int i=0;i<nums.length;++i){
            if(nums[i]<L && i>0)
                res+=dp;
            if(nums[i]>R){
                dp=0;
                prev=i;
            }
            if(L<=nums[i] && nums[i]<=R){
                dp=i-prev;
                res+=dp;
            }
        }
        return res;
    }
}

File: Solutions/795. Number of Subarrays with Bounded Maximum/795. Number of Subarrays with Bounded Maximum.go
func numSubarrayBoundedMax(nums []int, L int, R int) int {
    res,prev,dp:=0,-1,0
    for i:=0;i<len(nums);i+=1{
        if nums[i]<L && i>0{
            res+=dp
        }
        if nums[i]>R{
            dp=0
            prev=i
        }
        if L<=nums[i] && nums[i]<=R{
            dp=i-prev
            res+=dp
        }
    }
    return res
}